### PR TITLE
Example stream that copies documents into a new index.

### PR DESCRIPTION
**Pull request by steveblackmon**
For integration tests to work, the following need to merge into incubator-streams:
STREAMS-285
STREAMS-286

### CODE REVIEW

File: local/elasticsearch-reindex/src/main/java/org/apache/streams/elasticsearch/example/ElasticsearchReindex.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+    public ElasticsearchReindex(ElasticsearchReindexConfiguration reindex) {
+        this.reindex = reindex;
+    }
+
+    public static void main(String[] args)
+    {
+        LOGGER.info(StreamsConfigurator.config.toString());
+
+        executor.submit(new ElasticsearchReindex());
```
Comment (mfranklin) [inline local/elasticsearch-reindex/src/main/java/org/apache/streams/elasticsearch/example/ElasticsearchReindex.java]: Why run this in a separate thread?

File: local/elasticsearch-reindex/src/main/resources/log4j.properties [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Root logger option
```
Comment (mfranklin) [inline local/elasticsearch-reindex/src/main/resources/log4j.properties]: Shouldn't log4j be omitted from the classpath if logback is going to be the SLF4J backing implementation?

File: local/elasticsearch-reindex/src/main/java/org/apache/streams/elasticsearch/example/ElasticsearchReindex.java (hunk 1 above)
Comment (steveblackmon) [inline local/elasticsearch-reindex/src/main/java/org/apache/streams/elasticsearch/example/ElasticsearchReindex.java]: By implementing Runnable and placing all of the business logic outside the main method, it's easy to run the stream from other classes without dominating the stack of that class.  for example, note how ElasticsearchReindexParentChildIT creates and executes two streams in sequence.

**Comment by steveblackmon** [comment]
slf4j-log4j12 is actually banned by enforcer-plugin.  there are certain cases where setting log4j properties is helpful even when the app prefers slf4j-logback, typically where dependencies are logging with log4j and supplying a logback configuration file isn't getting the desired result.  that said i don't know that this example really needs this file.  will remove it to keep the example as minimal as possible.

File: local/elasticsearch-reindex/src/main/java/org/apache/streams/elasticsearch/example/ElasticsearchReindex.java (hunk 1 above)
Comment (mfranklin) [inline local/elasticsearch-reindex/src/main/java/org/apache/streams/elasticsearch/example/ElasticsearchReindex.java]: I completely agree on separating business logic from the main method.  You can do that and still get the desired result without spinning up an executor service in the main method.  Just call run directly.  Unless I am missing something??

**Comment by mfranklin** [comment]
Banning slf4j-log4j12 is an excellent idea, as you only want one backing implementation for logging.  I would ban all the other backing implementations as well, such as log4j.jar.  

However, you SHOULD include the bridges, so that logback is the only thing that has to be configured [1].  The bridges implement other logging APIs and delegate to SLF4J, which in turn delegates to an implementation; logback in this case. This means you can include dependencies that log to Log4j APIs, JUL, JCL, without having to include their configuration or backing implementations.

[1] http://www.slf4j.org/legacy.html

File: local/elasticsearch-reindex/src/main/java/org/apache/streams/elasticsearch/example/ElasticsearchReindex.java (hunk 1 above)
Comment (steveblackmon) [inline local/elasticsearch-reindex/src/main/java/org/apache/streams/elasticsearch/example/ElasticsearchReindex.java]: In the interest of keeping the example as simple as possible I removed the executor service.

**Comment by steveblackmon** [comment]
added more logging libs to imports and to the banned list

**Comment by mfranklin** [comment]
:+1:

**Pull request merged**